src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PreferIsExactSpellingIsTrueForKnownApisAnalyzer.cs

```diff
@@ -0,0 +1,164 @@
+            {
+                if (ctx.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesDllImportAttribute, out var dllImportType))
+                {
+                    ctx.RegisterSymbolAction(x => AnalyzeExternMethod(x, dllImportType), SymbolKind.Method);
+                }
+            });
+        }
+
+        private static void AnalyzeExternMethod(SymbolAnalysisContext context, INamedTypeSymbol dllImportType)
+        {
+            var methodSymbol = (IMethodSymbol)context.Symbol;
+            if (!methodSymbol.IsExtern)
+            {
+                return;
+            }
+
+            var dllImportAttribute = methodSymbol
+                .GetAttributes()
+                .FirstOrDefault(x => x.AttributeClass.Equals(dllImportType));
+
+            if (dllImportAttribute is null)
+            {
+                return;
+            }
+
+            var hasExactSpellingParameter = dllImportAttribute.NamedArguments.FirstOrDefault(x => x.Key.Equals("ExactSpelling", StringComparison.Ordinal));
+            var hasCharSetParameter = dllImportAttribute.NamedArguments.FirstOrDefault(x => x.Key.Equals("CharSet", StringComparison.Ordinal));
+
+            var methodName = methodSymbol.GetActualExternName(dllImportAttribute);
+            var isExactSpelling = hasExactSpellingParameter.Key is not null
+                && hasExactSpellingParameter.Value.Kind != TypedConstantKind.Array
+                && bool.TryParse(hasExactSpellingParameter.Value.Value?.ToString(), out var isExact)
+                && isExact;
+            var isCharSetUnicode = hasCharSetParameter.Key is not null
+                && hasCharSetParameter.Value.Kind != TypedConstantKind.Array
+                && Enum.TryParse<CharSet>(hasCharSetParameter.Value.Value?.ToString(), out var actualCharSet)
+                && actualCharSet == CharSet.Unicode;
+            var dllName = dllImportAttribute.ConstructorArguments.FirstOrDefault().Value?.ToString() ?? string.Empty;
+
+            if (!KnownApis.Value.TryGetValue(dllName, out var methods))
+            {
+                return;
+            }
+
+            if (methods.Contains(methodName))
+            {
+                if (isCharSetUnicode && !isExactSpelling && methodName.EndsWith("W", StringComparison.Ordinal))
+                {
+                    var diagnostic = methodSymbol.CreateDiagnostic(WideRule, methodName);
+                    context.ReportDiagnostic(diagnostic);
+                    return;
+                }
+
+                if (!isCharSetUnicode && !isExactSpelling && methodName.EndsWith("A", StringComparison.Ordinal))
+                {
+                    var diagnostic = methodSymbol.CreateDiagnostic(DefaultRule, methodName);
+                    context.ReportDiagnostic(diagnostic);
+                    return;
+                }
+            }
+
+            if (methods.Contains(methodName + "A") && !isCharSetUnicode)
```

Comment:
I think this should be within the search condition above (row 138 `if (methods.Contains(methodName))`) and seems it is missing scenarios with `W` suffix @jkotas [suggested ](https://github.com/dotnet/runtime/issues/35695#issuecomment-686663918):
> When we see "Xxxx" in the entrypoint, the ExactSpelling autofixer needs to decide whether to suggest:
> 
> "XxxxW" + ExactSpelling=true. This should be suggested when we see "XxxxW" in the database.
> "Xxxx" + ExactSpelling=true. This should be suggested when we see "Xxxx" and no "XxxxW" in the database
> Leave the API alone. This should be the case when neither "Xxxx" nor "XxxxW" is in the database

0. If `methodName` exist in DB (already covered with  `if (methods.Contains(methodName))` condition):
1. If `methodName + "W"` exist  we suggest adding `"W"` suffix and setting `ExactSpelling=true`
2. If `methodName + "W"` does not exist we suggest only setting `ExactSpelling=true`

Probably we want to do same for "A" suffix but from from @elinor-fung's [comment ](https://github.com/dotnet/runtime/issues/35695#issuecomment-623803330) seems suggesting `ExactSpelling=true` when `CharSet=CharSet.Ansi` might not important or might not much helpful, if we want to cover it the rule should be same

3. If `methodName + "A"` exist  we suggest adding `"A"` suffix and setting `ExactSpelling=true`
4. If `methodName + "A"` does not exist we suggest only setting `ExactSpelling=true`